packages/test/test-end-to-end-tests/src/test/gc/gcDatastoreAliased.spec.ts

```diff
@@ -59,12 +58,42 @@ describeFullCompat("GC Data Store Aliased", (getTestObjectProvider) => {
         await provider.ensureSynchronized();
 
         // Should be able to retrieve root datastore from remote
-        const containerRuntime2 = mainDataStore2._context.containerRuntime as IContainerRuntime;
+        const containerRuntime2 = mainDataStore2._context.containerRuntime as unknown as IContainerRuntime;
         assert.doesNotThrow(async () => containerRuntime2.getRootDataStore(alias),
             "Aliased datastore should be root as it is aliased!");
         summaryWithStats = await waitForSummary(container2);
-        gcState = getGCStateFromSummary(summaryWithStats.summary);
-        assert(gcState?.gcNodes[ds1.handle.absolutePath].unreferencedTimestampMs === undefined,
+        const gcStatePostAlias = getGCStateFromSummary(summaryWithStats.summary);
+        assert(gcStatePostAlias !== undefined, "Should get gc post state from summary!");
+        assert(gcStatePostAlias.gcNodes[ds1.handle.absolutePath].unreferencedTimestampMs === undefined,
             "AliasableDataStore1 should be referenced as it is aliased and thus a root datastore!");
     });
 });
+
+describeNoCompat("GC aliased datastore is always referenced", (getTestObjectProvider) => {
+    let provider: ITestObjectProvider;
+
+    beforeEach(async () => {
+        provider = getTestObjectProvider({ syncSummarizer: true });
+    });
+
+    it("GC aliased datastore is always referenced", async () => {
+        const container = await provider.makeTestContainer(defaultGCConfig);
+        const remoteContainer = await provider.loadTestContainer(defaultGCConfig);
+        await waitForContainerConnection(container);
+        await waitForContainerConnection(remoteContainer);
+
+        // Create and alias datastore
+        const mainDatastore = await requestFluidObject<ITestDataObject>(container, "default");
+        const aliasedDataStore = await mainDatastore._context.containerRuntime.createDataStore(TestDataObjectType);
+        const alias = "alias";
+        await aliasedDataStore.trySetAlias(alias);
```

Comment:
Do you think we should test scenarios where handle to aliased data store is added / removed. Basically, if the handle to an aliased data store is removed, it should still be referenced.

Reply:
Yeah, lets add that